entrypoint.sh

```diff
@@ -100,37 +99,139 @@ function super_lint() {
+    eval "$cmd"
+  else
+    for file in $(get_files_matching_filetype "${linter[filetype]}" "${included[@]}"); do 
+      if [[ "${linter[executor]+x}" ]]; then
+        cmd="${linter[executor]} -c \"${linter[name]} ${linter[args]} ${file}\""
+      else
+        cmd="${linter[name]} ${linter[args]} ${file}"
+      fi
+      eval "$cmd"
+    done
+  fi
+}
+
 function seiso_lint() {
+  echo -e "\nRunning Seiso Linter\n--------------------------\n"
+
   excluded=()
   included=()
 
   while read -r file; do
-    # Apply filter with =~ to ensure it is aligned with github/super-linter
     if [[ -n ${INPUT_EXCLUDE:+x} && "${file}" =~ ${INPUT_EXCLUDE} ]]; then
       excluded+=("${file}")
       continue
     fi
-
     included+=("${file}")
+  done < <(find . -path "./.git" -prune -or -type f)
+  
+  declare -gA linter_exit_codes
+  declare -gA pids
 
-    # Check Dockerfiles
-    if [[ "${file}" = *Dockerfile ]]; then
-      dockerfile_lint -f "${file}" -r /etc/opt/goat/oci.yml
-    fi
+  input="/etc/opt/goat/linters.json"
 
-    # Check .md file spelling and links
-    if [[ "${file}" = *.md ]]; then
-      npx cspell -c /etc/opt/goat/cspell.config.js -- "${file}"
-      npx markdown-link-check --config /etc/opt/goat/links.json --verbose "${file}"
+  while read -r line; do
+    if [[ $line == \#* ]]; then
+      continue
     fi
-  done < <(find . -path "./.git" -prune -or -type f)
-
-  echo "Scanned ${#included[@]} files"
-  echo "Excluded ${#excluded[@]} files"
+    
+    unset linter
+    declare -gA linter
+ 
+    while IFS='=' read -r key value; do
+      value=$(echo "$value" | tr -d "'")
+      linter["$key"]=$value
+    done < <(echo "$line" | jq -r 'to_entries|map("\(.key)=\(.value|tostring)")|.[]')
+        
+    linter+=([logfile]="/etc/opt/goat/logs/${linter[name]}.log")
+
+    echo "===============================" >> "${linter[logfile]}"
+    echo "Running linter: ${linter[name]}"
+    echo "${linter[name]^^}" >>"${linter[logfile]}"
+
+    lint_files & 
+    pid=$!
+    pids["$pid"]+="${linter[name]}"
+
+    echo "-------------------------------" >> "${linter[logfile]}"
+  done < <(jq -c '.[]' $input)
+
+  for p in "${!pids[@]}"; do
+    set +e
+    wait "$p"
+    exit_code=$?
+    linter_failed "$exit_code" "${pids[${p}]}"
+    set -e
+  done
+
+  cat /etc/opt/goat/logs/*
+  
+  echo -e "\nScanned ${#included[@]} files"
+  echo -e "Excluded ${#excluded[@]} files\n"
 }
 
+linter_failed="false"
 
 setup_environment
 check_environment
-super_lint
+
+set +e
+super_linter_result=$(super_lint >/dev/null; echo $?)
+if [ "${super_linter_result}" -ne 0 ]; then
+  linter_failed="true"
+fi
+set -e
+
 seiso_lint
+
+for lint in "${!linter_exit_codes[@]}"; do
+  if [[ "${linter_exit_codes[${lint}]}" -gt 0 ]]; then
+    feedback_label="ERROR"
+    linter_failed="true"
+  else
+    feedback_label="DEBUGGING"
+  fi
+
+  feedback "${feedback_label}" "${lint} discovered errors."
+done
+
+if [ "${linter_failed:-false}" == "true" ]; then
```

Comment:
We should always default closed, meaning if linter_failed was for whatever reason not set, it would fail the pipeline.